csunplugged/tests/resources/views/test_piano_keys.py

```diff
@@ -123,3 +101,18 @@ def test_piano_keys_resource_generation_missing_header_text_parameter(self):
             response.get("Content-Disposition"),
             'attachment; filename="{}"'.format(filename)
         )
+
+    def subtitle(self, combination):
```

Comment:
Are these exact copies of the subtitle function on the ResourceGenerator subclass?

Reply:
Pretty much, sometimes simplified. We could call the method directly, and write other test cases for testing the subtitle method returns what we expect?

Reply:
I'd much prefer to call method directly - this currently isn't testing anything (you're testing two copies of the same function produce the same result) - so either we should call the method directly, or just ignore the subtitle in the actual test. And then, as you say, test the subtitle method separately.